### ISSUE TITLE
Trivy config k8s manifests wrong lines with multiple objects in the same file

### ISSUE THREAD

**Issue by jesusfcr** [aquasecurity/trivy#2965]
## Description

When looking for misconfigurations in k8s yaml files with multiple object definition in the same file the CauseMetadata line numbers are wrong.
They are always based on the position inside the object assuming it is at the beginning of the file

## What did you expect to happen?

With this file `test.yaml`

```yaml
---
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: myrelease-comp1
spec:
  rules:
    - host: comp1.example.com
      http:
        paths:
          - path: /
            backend:
              serviceName: myrelease-comp1
              servicePort: 80
---
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: myrelease-comp2
spec:
  rules:
    - host: comp2.example2.com
      http:
        paths:
          - path: /
            backend:
              serviceName: myrelease-comp2
              servicePort: 80
```

The report should indicate the exact line for each object
First in `test.yaml:2-6` and the second in  `test.yaml:16-20`

## What happened instead?

Trivy identified both misconfigs in the same position `test.yaml:1-5`

See the output below.

## Output of run with `-debug`:

```
❯ trivy config . --debug
2022-09-30T12:30:56.722+0200	DEBUG	Severities: ["UNKNOWN" "LOW" "MEDIUM" "HIGH" "CRITICAL"]
2022-09-30T12:30:56.772+0200	DEBUG	cache dir:  /Users/jesus.carpintero/Library/Caches/trivy
2022-09-30T12:30:56.772+0200	DEBUG	Module dir: /Users/jesus.carpintero/.trivy/modules
2022-09-30T12:30:56.772+0200	INFO	Misconfiguration scanning is enabled
2022-09-30T12:30:57.043+0200	DEBUG	OS is not detected.
2022-09-30T12:30:57.043+0200	INFO	Detected config files: 1
2022-09-30T12:30:57.043+0200	DEBUG	Scanned config file: test.yaml

test.yaml (kubernetes)

Tests: 80 (SUCCESSES: 78, FAILURES: 2, EXCEPTIONS: 0)
Failures: 2 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 2, CRITICAL: 0)

HIGH: apiVersion 'networking.k8s.io/v1beta1' and kind ‘Ingress' should be replaced with the new API ‘networking.k8s.io.v1.Ingress'
═══════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════
apiVersion 'networking.k8s.io/v1beta1' and kind 'Ingress' has been deprecated on: 'v1.19' and planned for removal on:'v1.22'

See https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/api/networking/v1beta1/zz_generated.prerelease-lifecycle.go
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 test.yaml:1-5
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1 ┌ ---
   2 │ apiVersion: networking.k8s.io/v1beta1
   3 │ kind: Ingress
   4 │ metadata:
   5 └   name: myrelease-comp1
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────


HIGH: apiVersion 'networking.k8s.io/v1beta1' and kind ‘Ingress' should be replaced with the new API ‘networking.k8s.io.v1.Ingress'
═══════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════
apiVersion 'networking.k8s.io/v1beta1' and kind 'Ingress' has been deprecated on: 'v1.19' and planned for removal on:'v1.22'

See https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/api/networking/v1beta1/zz_generated.prerelease-lifecycle.go
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 test.yaml:1-5
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1 ┌ ---
   2 │ apiVersion: networking.k8s.io/v1beta1
   3 │ kind: Ingress
   4 │ metadata:
   5 └   name: myrelease-comp1
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
```

## Output of `trivy -v`:

```
Version: 0.32.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2022-09-29 12:08:18.974080174 +0000 UTC
  NextUpdate: 2022-09-29 18:08:18.974079574 +0000 UTC
  DownloadedAt: 2022-09-29 12:31:51.70267 +0000 UTC
```

**Comment by jesusfcr** [comment]
I identified where the code responsible for this reside and I opened https://github.com/aquasecurity/defsec/pull/1013

**Comment by jesusfcr** [comment]
Fixed bumping defsec 0.28.0 https://github.com/aquasecurity/defsec/releases/tag/v0.82.0
